api-definitions.ts

```diff
@@ -0,0 +1,265 @@
+    rename board ✅
+    create column ✅
+    rename column ✅
+    move column ✅
+    delete column ✅
+    create card ✅
+    move card ✅
+    delete card ✅
+
+    Actions to support later: 
+    Move cards between boards
+    create board
+    delete board
+
+	**Decisions**
+	 - Rename old Board entities to SingleColumnBoard to avoid conflicts
+
+    Other thoughts:
+        - we have to support cards being added/removed from other places (task overview)
+        - we have to support cards being added/removed from legacy (tasks and lessons)
+        - we might want to load board information from rough to detailed (we had the idea to first render loading states for all cards, and then query each card individually)
+        - create and delete board is not necessarily needed in the first version
+        - single requests for moving single elements, or reorder?
+
+*/
+
+/*
+  Regarding Multi-Tenant conflicts: 
+
+  Util we have Websockts to keep all clients (almost) in sync with each other, we have to deal with Conflicts in some way.
+
+  Option 1: We track a Board-Version for every mutation to the board and return a `409 - Conflict` on version mismatch. 
+            When there is a mismatch the board gets refreshed.
+
+  Option 2: We track a Board-Version for each board mutation. When there is a Version mismatch greater than +1 
+            (e.g. last known version is 10, backend returns a 12) the Frontend reloads the board.
+
+
+  **Intermediary decision**
+  Option 3: We ignore multi-tenant edits until we have websockts. 
+            On each board-mutation, we return the full board-skeleton and update to minimize potential conflicts.			
+*/
+
+type BoardCardType =
+	| "task"
+	| "content"
+	| "legacy-task-reference"
+	| "legacy-lesson-reference";
+
+// GET - Load a Board
+// /boards/:id
+interface Board {
+	id: string;
+	title: string;
+	columns: BoardColumn[];
+	timestamps: BoardTimestamps;
+}
+
+interface BoardTimestamps {
+	lastUpdatedAt: string;
+	createdAt: string;
+	deletedAt: string;
+}
+
+interface BoardColumn {
+	id: string;
+	title: string;
+	cards: BoardSkeletonCard[];
+}
+
+interface BoardSkeletonCard {
+	id: string;
+	height: number;
+}
+
+// GET - Load a list of cards
+// /cards?ids=123,456,789
+
+// Response
+// list of BoardCard
+
+interface BoardCard {
+	id: string;
+	height: number;
+	elements: ContentElement[];
+	cardType: BoardCardType;
+	visibility: VisibilitySettings;
+	title: string;
+}
+
+interface VisibilitySettings {
+	publishedAt: string | undefined; // undefined = invisible | Date specifies the visibility from
+}
+
+declare type AnyBoardCard = BoardLegacyLessonCard | BoardLegacyTaskCard; // union of all BoardCard-Interfaces
+
+interface BoardLegacyLessonCard extends BoardCard {
+	cardType: "legacy-lesson";
+	lessonId: string;
+	elements: ContentElement[]; // narrowed down to allowed Elements: LegacyLesson
+
+	// some properties
+}
+
+interface BoardLegacyTaskCard extends BoardCard {
+	cardType: "legacy-task";
```

Comment:
```suggestion
	cardType: "legacy-task-reference";
```

Should it be legacy-task-reference?